crates/solver/src/liquidity/zeroex.rs

```diff
@@ -77,6 +92,50 @@ impl ZeroExLiquidity {
         };
         Some(Liquidity::LimitOrder(limit_order))
     }
+
+    async fn run_orderbook_fetching(
+        api: Arc<dyn ZeroExApi>,
+        blocks_stream: CurrentBlockStream,
+        orderbook_cache: Arc<OrderbookCache>,
+        gpv2_address: H160,
+    ) {
+        let mut receiver = blocks_stream.clone();
+        loop {
```

Comment:
I think the block stream yields the current block immediately when turned into a stream so you get the first execution of the loop right away on startup (and don't need the error handling below, because the stream should never end)

```suggestion
  let mut blocks_stream = into_stream(blocks_stream);
  while let Some(_) = blocks_stream.next().await {
```

Reply:
Updated https://github.com/cowprotocol/services/pull/2639/commits/6205a72259c7191a295d9085d0dd2bf156afd1b5

IMO, we still need to use logs to indicate any problems with the stream that either went into limbo or returned unexpected results(None).

Reply:
FIxed